cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableManager.java

```diff
@@ -524,6 +530,19 @@ private String generateFileSetCreateStatement(DatasetId datasetId, Dataset datas
     String databaseName = ExploreProperties.getExploreDatabaseName(properties);
     Map<String, String> tableProperties = FileSetProperties.getTableProperties(properties);
 
+    // if this dataset reuses an existing table, do not attempt to create it
+    if (FileSetProperties.isUseExisting(tableProperties)
+      || FileSetProperties.isPossessExisting(tableProperties)) {
+      try {
+        exploreService.getTableInfo(databaseName, tableName);
+        // table exists: do not attempt to create
+        return null;
+      } catch (Exception e) {
+        LOG.warn("Dataset '{}' is configured to use an existing explore table, but table '{}' does not " +
```

Comment:
Regarding the exception from getTableInfo()... I guess ExploreService should have a method tableExists(), but it does not. So I looked for other place in the code that need this and copy/pasted. The original code that does this is here: https://github.com/caskdata/cdap/blob/c021855d8cec1ea54549b1b4b901f88cf1b1d5a4/cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableManager.java#L301. Check out who wrote that :)

I would like to open a Jira to address this (add an exists method to explore service) so that we can get this one merged. Would that be ok?